drivers/fuzz/src/sha256.rs

```diff
@@ -0,0 +1,72 @@
+// Licensed under the Apache-2.0 license
+
+use caliptra_drivers::{Array4x8, CaliptraResult, Sha256};
+
+use sha2::digest::block_buffer::Block;
+use sha2::digest::consts::U64;
+
+const SHA256_BLOCK_BYTE_SIZE: usize = 64;
+
+#[derive(Default)]
+pub struct Sha256SoftwareDriver {
+    /// Hash
+    hash: [u32; 8],
+}
+
+impl Sha256 for Sha256SoftwareDriver {
+    unsafe fn zeroize() {
+        // No hardware registers to zero
+    }
+
+    fn zeroize_internal(&mut self) {
+        self.hash = Self::HASH_IV_256;
+    }
+
+    fn copy_digest_to_buf(&mut self, buf: &mut Array4x8) -> CaliptraResult<()> {
+        *buf = Array4x8::from(self.hash);
+        Ok(())
+    }
+
+    fn digest_block(
+        &mut self,
+        block: &[u8; SHA256_BLOCK_BYTE_SIZE],
+        _first: bool,
```

Comment:
Don't you need to reset the hash when _first is true?

There's no way this implementation is going to generate the correct output if it mixes ALL operations together.

Reply:
Oh. Possibly. Because while new Sha256DigestOp objects are created, there is only one driver passed to each. They'd have to share internal state. I'll probably need it to call zeroize(), like digest() does.

Reply:
The new abstraction just calls the high-level sha256 methods directly, so there's no need for this change.